src/app/dataset-view/additional-components/metadata-component/metadata.component.html

```diff
@@ -266,9 +276,114 @@
+                            <span class="position-absolute text-muted group-box-label text-uppercase">Preprocess</span>
+                            <div class="mx-4">
+                                <app-block-row-data
+                                    [tooltip]="'Identifier of the engine used for this transformation.'"
+                                    [label]="'Engine:'"
+                                >
+                                    <app-engine-property [data]="source.preprocess.engine"></app-engine-property>
+                                </app-block-row-data>
+                            </div>
+
+                            <div class="mx-4">
+                                <app-block-row-data
+                                    [tooltip]="'Queries use for specifying multi-step SQL transformations.'"
+                                    [label]="'Queries:'"
+                                >
+                                </app-block-row-data>
+                                <div class="row my-4">
+                                    <app-sql-query-viewer [data]="source.preprocess.queries"></app-sql-query-viewer>
+                                </div>
+                            </div>
+                        </div>
+
+                        <div class="position-relative border border-1 p-2 rounded-3 mt-4">
+                            <span class="position-absolute text-muted group-box-label text-uppercase">Merge</span>
+                            <div class="mx-4" *ngIf="source.merge.__typename">
+                                <app-block-row-data
+                                    [tooltip]="
+                                        'Merge strategy determines how newly ingested data should be combined with the data that already exists in the dataset.'
+                                    "
+                                    [label]="'Strategy:'"
+                                >
+                                    <app-merge-strategy-property
+                                        [data]="source.merge.__typename"
+                                    ></app-merge-strategy-property>
+                                </app-block-row-data>
+                            </div>
+
+                            <div class="mx-4" *ngIf="source.merge.__typename === 'MergeStrategyLedger'">
+                                <app-block-row-data
+                                    [tooltip]="
+                                        'Names of the columns that uniquely identify the record throughout its lifetime.'
+                                    "
+                                    [label]="'Primary key:'"
+                                >
+                                    <app-cards-property [data]="source.merge.primaryKey"></app-cards-property>
+                                </app-block-row-data>
+                            </div>
+                        </div>
+                    </div>
```

Comment:
This is again, very similar to the code in the same template that handles polling sources. Think of extracting shared part into one or more sub-components to avoid dumb copy pasting.

Reply:
Added onr common template